frontend/src/pages/admin/OverviewPage/components/CachingPanel.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+
+import { createNotification } from "@app/components/notifications";
+import { Button, DeleteActionModal } from "@app/components/v2";
+import { useOrgPermission } from "@app/context";
+import { usePopUp } from "@app/hooks";
+import { useInvalidateCache } from "@app/hooks/api";
+import { CacheType } from "@app/hooks/api/admin/types";
+
+export const CachingPanel = () => {
+  const { mutateAsync: invalidateCache } = useInvalidateCache();
+  const { membership } = useOrgPermission();
+
+  const [type, setType] = useState<CacheType | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { popUp, handlePopUpOpen, handlePopUpClose, handlePopUpToggle } = usePopUp([
+    "invalidateCache"
+  ] as const);
+
+  const handleInvalidateCacheSubmit = async () => {
+    if (!type) return;
+    setIsLoading(true);
+
+    try {
+      await invalidateCache({ type });
+
+      createNotification({
+        text: `Successfully invalidated ${type} cache`,
+        type: "success"
+      });
+
+      setType(null);
+      handlePopUpClose("invalidateCache");
+    } catch (err) {
+      console.error(err);
+      createNotification({
+        text: `Failed to invalidate ${type} cache`,
+        type: "error"
+      });
+    }
+
+    setIsLoading(false);
+  };
+
+  return (
+    <>
+      <div className="mb-6 flex flex-wrap items-end justify-between gap-4 rounded-lg border border-mineshaft-600 bg-mineshaft-900 p-4">
+        <div className="flex flex-col">
+          <span className="mb-2 text-xl font-semibold text-mineshaft-100">Secrets Cache</span>
+          <span className="max-w-xl text-sm text-mineshaft-400">
+            The secrets cache encompasses all secrets stored within the system and provides a
```

Comment:
Can we say `encrypted secrets`, just so that users feel confident 😅 on the cache